src/defaults/queue.js

```diff
@@ -5,8 +5,13 @@ function enqueue(array, item, context) {
 }
 
 function dequeue(array, item, context) {
-  const [, ...rest] = array;
-  return rest;
+  let index = array.indexOf(item.meta.offlineAction);
```

Comment:
@sorodrigo Is `item` guaranteed have `.offlineAction`? I thought COMMIT actions do not carry this?

If I do a:


```
            dispatch({
              type: 'FOLLOW_USER_REQUEST',
              payload: { userId },
              meta: {
                offline: {
                  // the network action to execute:
                  effect: { url: 'http://localhost:1337/api/follow', method: 'POST', json: { userId } },
                  // action to dispatch when effect succeeds:
                  commit: { type: 'FOLLOW_USER_COMMIT', meta: { userId } },
                  // action to dispatch if network action fails permanently:
                  rollback: { type: 'FOLLOW_USER_ROLLBACK', meta: { userId } }
                }
              }
            });
```

I then see this action:

<img width="405" alt="image" src="https://user-images.githubusercontent.com/6372489/64134096-d8a1a380-cd8f-11e9-83c7-59a7e75b7d18.png">

Reply:
@Noitidart That seemed odd so I had to check and we're only adding the offlineAction to the default actions. Can you confirm you're implementation depends on using defaultCommit @DianaKoenraadt ?

This to me seems wrong, I have to do some historian work to see if we made this decision on purpose or it's actually a bug. Thanks for pointing it out!

After much thought, I think this addition will have to be tested carefully. I see that the implementation relies on using `indexOf(offlineAction)` where offlineAction is an object. I'm not entirely sure this will work across sessions if the reference to the object changes (maybe it's always constant in this case), perhaps it's best to check  and dequeue using the transactionId.

Reply:
Thanks for those notes @sorodrigo! I recently joined a company that is using redux-offline, so you might be seeing me around here! :)